### PR TITLE
doc: openthread: Update OpenThread memory usage tables

### PR DESCRIPTION
**Pull request by pszkotak**
Add information about sample memory usage when building with
different variants of the OpenThread libraries.

Signed-off-by: Piotr Szkotak <piotr.szkotak@nordicsemi.no>

### CODE REVIEW

**Review by edmont — CHANGES_REQUESTED** [review]
The columns about available RAM and ROM seem to have wrong information, in some cases even more than the total memory :)

Also the tables look too big now, maybe we could split them in two? I.e., for every device one table with and without multiprotocol enabled, for instance.

**Comment by greg-fer** [comment]
Would it be ok to update the table look after the release, and use this PR only for memory values and fixing the term `memory space` (noticed by @frkv , discussed with @pszkotak )?

**Comment by edmont** [comment]
> Would it be ok to update the table look after the release, and use this PR only for memory values and fixing the term `memory space` (noticed by @frkv , discussed with @pszkotak )?

Sure.

File: doc/nrf/ug_thread_ot_memory.rst [hunk 1]
```diff
@@ -41,29 +41,45 @@ nRF52840 DK RAM and flash memory requirements
...
+| Role       | Sample                                  | Library variant   | Multiprotocol   |   ROM OpenThread stack + App [kB] |   ROM Bluetooth LE stack [kB] |   Persistent storage [kB] |   RAM OpenThread stack + App [kB] |   RAM Bluetooth LE stack [kB] |   Avaialble ROM [kB] |   Total ROM [kB] |   Available RAM [kB] |   Total RAM [kB] |
++============+=========================================+===================+=================+===================================+===============================+===========================+===================================+===============================+======================+==================+======================+==================+
+| Router     | :ref:`CLI <ot_cli_sample>`              | Master            | No              |                               448 |                             0 |                        32 |                                92 |                             0 |                  480 |             1024 |                   92 |              256 |
```
Comment (MarekPorwisz) [inline doc/nrf/ug_thread_ot_memory.rst:47]: I think, that Available RAM [kB] is in fact total RAM used at least it looks like it contains the sum of Thread stack ram usage and BT ram

**Review by MarekPorwisz — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_ot_memory.rst (hunk 1 above)
Comment (MarekPorwisz) [reply doc/nrf/ug_thread_ot_memory.rst:47]: Same goes for Available ROM

Comment (pszkotak) [reply doc/nrf/ug_thread_ot_memory.rst]: yes, fixed

**Comment by pszkotak** [comment]
> Would it be ok to update the table look after the release, and use this PR only for memory values and fixing the term `memory space` (noticed by @frkv , discussed with @pszkotak )?

Clarified the intended meaning in a separate commit.

**Review by edmont — APPROVED** [review]
My thinking was to _make the subtraction for the user_, i.e., for every building option show clearly:
- How much memory the device has.
- How much is taken by NCS.
- **How much is available for the user application**.

I believe the last point is the most interesting value for the user, and at the moment they need to infer it from the table. I would even remove the fixed memory sizes from the devices from the table and mention them on the top.

That said,  I agree those optimizations can be postponed after the release.

**Review by MarekPorwisz — APPROVED** [review]

**Review by frkv — APPROVED** [review]
LGTM

**Review by greg-fer — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**